File: Intermediate/10.Write a program to check if two strings are anagrams/Solution.py
def are_anagrams(str1, str2):
    if len(str1) != len(str2):
        return False

    return sorted(str1) == sorted(str2)

if __name__ == "__main__":
    str1 = input("Enter the first string: ")
    str2 = input("Enter the second string: ")

    if are_anagrams(str1, str2):
        print("The strings are anagrams.")
    else:
        print("The strings are not anagrams.")

File: Intermediate/10.Write a program to check if two strings are anagrams/Solution.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

void sortString(char* str) {
    int i, j;
    char temp;
    int len = strlen(str);

    for (i = 0; i < len - 1; i++) {
        for (j = i + 1; j < len; j++) {
            if (str[i] > str[j]) {
                temp = str[i];
                str[i] = str[j];
                str[j] = temp;
            }
        }
    }
}

bool areAnagrams(char* str1, char* str2) {
    if (strlen(str1) != strlen(str2))
        return false;

    sortString(str1);
    sortString(str2);

    return strcmp(str1, str2) == 0;
}

int main() {
    char str1[100], str2[100];

    printf("Enter the first string: ");
    scanf("%s", str1);
    printf("Enter the second string: ");
    scanf("%s", str2);

    if (areAnagrams(str1, str2))
        printf("The strings are anagrams.\n");
    else
        printf("The strings are not anagrams.\n");

    return 0;
}
